### ISSUE TITLE
connector/github: grab membership of orgs and/or teams instead of just teams

### ISSUE THREAD

**Issue by danielfm** [coreos/dex#691]
After reading the quick start guide (and the source code, since the documentation at this point is lacking), I managed to get the GitHub connector to work with the following configuration:

```yaml
# ./examples/config-dev.yaml
connectors:
- type: github
  id: github
  name: github
  config:
    redirectURI: http://127.0.0.1:5556/dex/callback
    clientID: ***
    clientSecret: ***
    org: my-org
```

With this, I can start the dex server and the example openid client and complete the authentication workflow:

```sh
# First terminal window
./bin/dex serve examples/config-dev.yaml

# Second terminal window
./bin/example-app
```

The only thing I missed was the ability to only authorize users from some organization. I noticed the source code has the concept of groups, but I tried to set the `org` configuration key but it doesn't seem to do anything.

Am I doing something wrong?

**Comment by ericchiang** [comment]
> and the source code, since the documentation at this point is lacking

Yep apologies, we're still working on finalizing the GitHub connector (and what a group means to it). You can track documentation progress for login strategies here: https://github.com/coreos/dex/issues/678

> The only thing I missed was the ability to only authorize users from some organization.

No connectors have the ability to restrict login to a subset of the users in a backing management system.

This has been brought up and we'd be open to someone putting together a more complete proposal. Perhaps a dup of https://github.com/coreos/dex/issues/66?

**Comment by danielfm** [comment]
@ericchiang sure, #66 seems to address what I described in this issue.

Sorry for the dup, feel free to close this!

**Comment by ericchiang** [comment]
@danielfm cool. A few other issues depend on this so I've add it to the "v2.0.0" [milestone](https://github.com/coreos/dex/milestones).

However I think this brings up a good point about organizations and groups. Since GitHub has two concepts of groups, orgs and teams, we probably want to be able to grab both. Right now we only do teams within an org.

Cool if I commandeer your issue for that issue? That is, rename this issue to be something like

```
connector/github: grab membership of orgs and/or teams instead of just teams
```

**Comment by danielfm** [comment]
Done!

**Comment by estroz** [comment]
Fixed by #1013
